### PR TITLE
- add gemmlowp/cci.20211003

### PR DESCRIPTION
**Pull request by SSE4**
Specify library name and version:  **gemmlowp/cci.20211003**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/gemmlowp/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+
+    def package_info(self):
+        self.cpp_info.components["eight_bit_int_gemm"].names["cmake_find_package"] = "eight_bit_int_gemm"
+        self.cpp_info.components["eight_bit_int_gemm"].names["cmake_find_package_multi"] = "eight_bit_int_gemm"
+        self.cpp_info.components["eight_bit_int_gemm"].libs = ["eight_bit_int_gemm"]
```
Comment (SSE4) [inline recipes/gemmlowp/all/conanfile.py:66]:
```suggestion
        self.cpp_info.components["eight_bit_int_gemm"].libs = ["eight_bit_int_gemm"]
        if self.settings.os == "Linux":
            self.cpp_info.components["eight_bit_int_gemm"].system_libs.extend(["pthread"])
```

File: recipes/gemmlowp/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
```
Comment (SSE4) [inline recipes/gemmlowp/all/conanfile.py:33]:
```suggestion

    def validate(self):
        if self.settings.os == "Windows" and self.options.shared:
            raise ConanInvalidConfiguration("shared is not supported on Windows")

```

File: recipes/gemmlowp/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import os
+from conans import ConanFile, CMake, tools
```
Comment (SSE4) [inline recipes/gemmlowp/all/conanfile.py:2]:
```suggestion
from conans import ConanFile, CMake, tools
from conans.errors import ConanInvalidConfiguration
```

File: recipes/gemmlowp/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        self._cmake.definitions['BUILD_TESTING'] = False
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("COPYING", dst="licenses", src=self._source_subfolder)
```
Comment (uilianries) [inline recipes/gemmlowp/all/conanfile.py:60]:
```suggestion
```

Only LICENSE is listed in project's folder

**Review by uilianries — DISMISSED** [review]

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — DISMISSED** [review]

**Review by jgsogo — CHANGES_REQUESTED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]

**Comment by SpaceIm** [comment]
Several issues in this recipe:

- upstream aways build shared lib
- include dir is wrong. In upstream CMake config file:

  ```cmake
  # Create imported target gemmlowp::gemmlowp
  add_library(gemmlowp::gemmlowp INTERFACE IMPORTED)
  
  set_target_properties(gemmlowp::gemmlowp PROPERTIES
    INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include/gemmlowp"
    INTERFACE_LINK_LIBRARIES "gemmlowp::eight_bit_int_gemm"
  )
  
  # Create imported target gemmlowp::eight_bit_int_gemm
  add_library(gemmlowp::eight_bit_int_gemm SHARED IMPORTED)
  
  set_target_properties(gemmlowp::eight_bit_int_gemm PROPERTIES
    INTERFACE_LINK_LIBRARIES "pthread"
  )
  ```

Less important:
- no share & pkgconfig dir to remove
- generators are useless (except cmake)
- required_conan_version is too high

**Pull request merged**